examples/platforms/virtual/logging.c

```diff
@@ -46,21 +46,41 @@
     offset += (unsigned int)charsWritten;                                    \
     VerifyOrExit(offset < sizeof(logString), logString[sizeof(logString) -1 ] = 0)
 
-void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, ...)
+#ifdef _WIN32
+void GetTimeString(char *timeString, size_t length)
+{
+    struct tm tmLocalTime;
+    time_t now = time(NULL);
+    (void)localtime_s(&tmLocalTime, &now);
+
+    strftime(timeString, length, "%Y-%m-%d %H:%M:%S ", &tmLocalTime);
+}
+#else
+void GetTimeString(char *timeString, size_t length)
 {
     struct timeval tv;
-    char timeString[40];
+    size_t offset;
+
+    gettimeofday(&tv, NULL);
+
+    offset = strftime(timeString, length, "%Y-%m-%d %H:%M:%S", localtime(&tv.tv_sec));
+    snprintf(&timeString[offset], length - offset, ".%06d", (uint32_t)tv.tv_usec);
+}
+#endif
+
+void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, ...)
+{
+    char timeString[50];
```

Comment:
I'd be inclined to implement `gettimeofday()` wrapper on Windows and get rid of this preprocessor check.

This pattern is summed up best [here](https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Generic-Functions.html):

> Autoconf follows a philosophy that was formed over the years by those who have struggled for portability: isolate the portability issues in specific files, and then program as if you were in a Posix environment. Some functions may be missing or unfixable, and your package must be ready to replace them.

However, I don't think we have any good pre-existing examples in the code base on how to do that. Will think about this some more.